File: backend/models/FlightSeat.js
const mongoose = require("mongoose");

const flightSeatSchema = new mongoose.Schema({
    seatId: { type: String, required: true, unique: true },
    flightId: { type: String, required: true },
    seatNumber: { type: String, required: true },
    rowNumber: { type: Number, required: true },
    seatColumn: { type: String, required: true },
    cabinClass: { type: String, required: true }, // "ECONOMY", "BUSINESS"
    seatStatus: { type: String, default: "AVAILABLE" }, // "AVAILABLE", "OCCUPIED", "HELD"
    holdToken: { type: String, default: null },
    holdExpiresAt: { type: Date, default: null }
});

module.exports = mongoose.model("FlightSeat", flightSeatSchema, "flight_seats");

File: app/src/main/java/com/skyline/app/network/FlightSeat.java
package com.skyline.app.network;

public class FlightSeat {
    private String seatId;
    private String seatNumber;
    private int rowNumber;
    private String seatColumn;
    private String cabinClass;
    private String seatStatus; // "AVAILABLE", "OCCUPIED"

    public String getSeatNumber() { return seatNumber; }
    public int getRowNumber() { return rowNumber; }
    public String getSeatColumn() { return seatColumn; }
    public String getCabinClass() { return cabinClass; }
    public String getSeatStatus() { return seatStatus; }
}